Texture.go

```diff
@@ -19,25 +19,31 @@ type loadImageResult struct {
 	err error
 }
 
-// Create new texture from rgba.
-// Note: this function has to be invokded in a go routine.
-// If call this in mainthread will result in stuck.
-func NewTextureFromRgba(rgba image.Image) (*Texture, error) {
-	Update()
-	result := mainthread.CallVal(func() interface{} {
-		texId, err := Context.renderer.LoadImage(ImageToRgba(rgba))
-		return &loadImageResult{id: texId, err: err}
-	})
+// NewTextureFromRgba creates a new texture from image.Image and, when it is done, calls loadCallback(loadedTexture).
+func NewTextureFromRgba(rgba image.Image, loadCallback func(*Texture)) {
+	go func() {
+		Update()
+		result := mainthread.CallVal(func() interface{} {
+			texId, err := Context.renderer.LoadImage(ImageToRgba(rgba))
+			return &loadImageResult{id: texId, err: err}
+		})
+
+		tid, ok := result.(*loadImageResult)
+		switch {
+		case !ok:
+			panic("giu: NewTextureFromRgba: unexpected error occured")
+		case tid.err != nil:
+			panic(fmt.Sprintf("giu: NewTextureFromRgba: error loading texture: %v", tid.err))
+		}
```

Comment:
@AllenDang is there any commonly known (expected) case when this may happen?
I mean if calling panic is ok here or should I handle it in any other way?

Reply:
@gucio321 I think panic is ok for now.